src/pages/workspace/WorkspaceInvitePage.js

```diff
@@ -65,93 +64,71 @@ const defaultProps = {
     ...policyDefaultProps,
 };
 
-class WorkspaceInvitePage extends React.Component {
-    constructor(props) {
-        super(props);
-
-        this.inviteUser = this.inviteUser.bind(this);
-        this.clearErrors = this.clearErrors.bind(this);
-        this.getExcludedUsers = this.getExcludedUsers.bind(this);
-        this.toggleOption = this.toggleOption.bind(this);
-        this.updateOptionsWithSearchTerm = this.updateOptionsWithSearchTerm.bind(this);
-        this.openPrivacyURL = this.openPrivacyURL.bind(this);
-
-        const {personalDetails, userToInvite} = OptionsListUtils.getMemberInviteOptions(props.personalDetails, props.betas, '', this.getExcludedUsers());
-        this.state = {
-            searchTerm: '',
-            personalDetails,
-            selectedOptions: [],
-            userToInvite,
-        };
-    }
-
-    componentDidMount() {
-        this.clearErrors();
-        const policyMemberEmailsToAccountIDs = PolicyUtils.getClientPolicyMemberEmailsToAccountIDs(this.props.policyMembers, this.props.personalDetails);
-        Policy.openWorkspaceInvitePage(this.props.route.params.policyID, _.keys(policyMemberEmailsToAccountIDs));
-    }
-
-    componentDidUpdate(prevProps) {
-        if (!_.isEqual(prevProps.personalDetails, this.props.personalDetails) || !_.isEqual(prevProps.policyMembers, this.props.policyMembers)) {
-            this.updateOptionsWithSearchTerm(this.state.searchTerm);
-        }
-
-        const isReconnecting = prevProps.network.isOffline && !this.props.network.isOffline;
-        if (!isReconnecting) {
+function WorkspaceInvitePage(props) {
+    const [searchTerm, setSearchTerm] = useState('');
+    const [selectedOptions, setSelectedOptions] = useState([]);
+    const [personalDetails, setPersonalDetails] = useState([]);
+    const [userToInvite, setUserToInvite] = useState(null);
+    const openWorkspaceInvitePageRef = useRef(null);
+
+    openWorkspaceInvitePageRef.current = () => {
+        const policyMemberEmailsToAccountIDs = PolicyUtils.getClientPolicyMemberEmailsToAccountIDs(props.policyMembers, props.personalDetails);
+        Policy.openWorkspaceInvitePage(props.route.params.policyID, _.keys(policyMemberEmailsToAccountIDs));
+    };
+
+    useEffect(() => {
+        Policy.clearErrors(props.route.params.policyID);
+        openWorkspaceInvitePageRef.current();
+    }, [props.route.params.policyID]);
```

Comment:
Not sure why need this useEffect?

Reply:
If this is for componentDidMount should go [] arguments.

Reply:
This is more extensive over-engineering, I don't see any use case. So we should just stick to the basics here.

Reply:
Adding dependencies to the dependency list is the basics, I think.

Reply:
Is there a valid case for this Suppose that a user goes to route with policyID = a from route with policyID = b by any link. ?

Reply:
For `params.policyID` changes, the component will be remounted. So I don't see any case where we would need this. If there is such a case please explain, if not please remove it thanks!

Reply:
I think `no`. But it may be later and also safe. No side effects

Reply:
Why is that remounted? client-side routing can avoid remounting

Reply:
If you still want to empty the dependency, I'll do

Reply:
> I think not. But it may be later and also safe. No side effects

Please remove it for now, as it's not required. For later needs will update when it's required, thanks!

Reply:
done